woocommerce/woocommerce-ios
# Dynamic Dashboard: Reviews Card Reload Functionality

Part of: #12742


## Description


This PR includes:
- initial logic for fetching reviews (and products), and displaying it on the UI.
- functionality for viewing all reviews

The main purpose of this PR is to get more eyes on the reviews and products fetching logic, and make sure the logic is sound, before going forward with more functionality. Because of that, some functionalities are missing and will be added in subsequent PR:
1. The card doesn't load notifications yet, which is needed to determine balloon color (read or not read). The loading logic should be similar to products loading, so should be relatively simple to add once the products loading is confirmed.
2. Filtering is not done yet. Still discussing in Slack what's the best way forward.
3. Tapping each review is not added yet.
4. Empty and error states are not added yet.

## Testing instructions


1. Run app, use a test site with at least 1 review on it.
5. Enable the "Most recent reviews" card in the dashboard if you haven't,
6. wait and ensure it loads and shows three reviews, with the correct subject, review content, and star.
7. Try the "View all reviews" and ensure it opens the Reviews list screen,
8. Go back to Dashboard and ensure the card still shows 3 cards as before.

## Screenshots


<img src="https://github.com/woocommerce/woocommerce-ios/assets/266376/fc8609dc-bafd-4a31-ab39-092d5b748f2c" width="42%">
